File: python/song.py
import chord_tree
import os
import time
import chord

class Song(object):
    """docstring for Song"""
    def __init__(self, difficulty):
        self.difficulty = difficulty
        self.potential_chords = chord_tree.PotentialChords()
        self.chords = []
        while self.is_over() != True:
            self.get_chord()

    def get_chord(self):
        if self.difficulty == 'hard':
            self.chords.append(self.potential_chords.random_chord())
        else:
            self.chords.append(self.potential_chords.random_easy_chord())

    def length(self):
        return len(self.chords)

    def is_over(self):
        return self.length() > 5
        # and self.chords[-1].eq_to(self.chords[0])

    def render(self):
        for chord in self.chords:
            os.system('clear')
            chord.render()
            time.sleep(1)

File: javascript/song.js
const PotentialChords = require('./chord_tree.js');

const Song = function (difficulty) {
  this.difficulty = difficulty;
  this.potentialChords = new PotentialChords();
  this.chords = [];

  while (!this.isOver()) {
    this.getChord();
  };
};

Song.prototype.getChord = function () {
  if (this.difficulty === 'hard') {
    this.chords.push(this.potentialChords.randomChord());
  } else {
    this.chords.push(this.potentialChords.randomEasyChord());
  }
};

Song.prototype.length = function () {
  return this.chords.length;
};

Song.prototype.isOver = function () {
  return this.chords.length > 5;
  //   && this.chords[-1].eqTo(this.chords[0])
};

Song.prototype.render = function (completionCallback) {
  console.log(this.chords);
  this.chords.forEach( (chord, i) => {
    const endIdx = this.chords.length - 1;
    setTimeout(() => {
      console.log('\033c');
      chord.render();
        if (i == endIdx) {
          console.log('we`re at the end!' );
          completionCallback()
        };
      }, 1000*i)
  });
};

module.exports = Song;
